cmd/tcpgoon.go

```diff
@@ -101,5 +101,5 @@ func run(params tcpgoonParams) {
 	mtcpclient.MultiTCPConnect(params.numberConnections, params.delay, params.target, params.port, connStatusCh, closureCh)
 	fmt.Fprintln(debugging.DebugOut, "Tests execution completed")
 
-	cmdutil.CloseNicely(params.targetip, params.target, params.port, connStatusTracker)
+	cmdutil.CloseNicely(params.targetip, params.target, params.port, *connStatusTracker)
```

Comment:
? I don't see changes in this method signature, so I don't understand why you are passing *connStatusTracker rather than connStatusTracker

Reply:
because, from in this branch, I transformed connStatusTracker to a pointer so I could update metrics

Reply:
ok, the signature did not change in fact :/ (it did sound strange to me the closure method updating the metrics) . You are passing the value of the pointer, so no changes on this, but you are storing the reference so other functions can update the contents of this struct.